app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+
+  useEffect(() => {
+    if (isNameError) {
+      setIsSaveDisable(true);
+    }
+  }, [isNameError]);
+
+  useEffect(() => {
+    setRecipeFormData(recipeData);
+  }, [recipeData]);
+
+  useEffect(() => {
+    if (
+      recipeFormData.recipeName === '' ||
+      recipeFormData.description === '' ||
+      recipeFormData.recipeName?.trim().length === 0 ||
+      recipeFormData.description?.trim().length === 0
+    ) {
+      setIsSaveDisable(true);
+    } else {
+      setIsNameError(false);
+      setIsSaveDisable(false);
+    }
+  }, [recipeFormData]);
+
+  const onCancel = () => {
+    setIsCreateAndEditRecipeFormOpen(false);
+    setIsNameError(false);
+  };
+
+  const onFormHandle = (event: FormEvent) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeData) => {
+    if (recipeFormAction === EditRecipeFormAction) {
+      const params = {
+        context: getCurrentNamespace(),
+        recipe_id: recipeData.recipeId.recipeId,
+      };
+      const payload = {
+        recipeName: recipeFormData.recipeName,
+        description: recipeFormData.description,
+        directives: recipeFormData.directives,
+      };
+      MyDataPrepApi.updateRecipe(params, payload).subscribe(
+        () => {
+          setIsNameError(false);
+          setIsCreateAndEditRecipeFormOpen(false);
+          setSnackbar({
+            open: true,
+            isSuccess: true,
+            message: `${recipeSteps.length} ${T.translate(
+              'features.WranglerNewUI.RecipeForm.labels.recipeSaveSuccessMessage'
+            )}`,
+          });
+        },
+        (err) => {
+          if (err.response.message) {
+            setIsNameError(true);
+          } else {
+            setIsCreateAndEditRecipeFormOpen(true);
+            setSnackbar({
+              open: true,
+              isSuccess: false,
+              message: T.translate(
+                'features.WranglerNewUI.RecipeForm.labels.errorMessage'
+              ).toString(),
+            });
+          }
+        }
+      );
+    }
+  };
+
+  return (
+    <StyledForm
+      onSubmit={(event: FormEvent) => onFormHandle(event)}
+      data-testid="recipe-form-parent"
+    >
+      <FormFieldWrapper>
+        <StyledLabel data-testid="recipe-name-label">
+          {T.translate('features.WranglerNewUI.RecipeForm.labels.name')}
+        </StyledLabel>
+        <StyledRecipeNameTextField
+          required
+          variant="outlined"
+          defaultValue={recipeData.recipeName}
+          error={isNameError}
+          id="outlined-error-helper-text"
+          helperText={
+            isNameError
+              ? T.translate('features.WranglerNewUI.RecipeForm.labels.nameErrorMessage')
+              : ''
+          }
+          fullWidth
+          onChange={(event) =>
+            setRecipeFormData({ ...recipeFormData, ['recipeName']: event.target.value })
+          }
```

Comment:
this should be simplified to `onChange={onRecipeNameChange}`

Reply:
now using onChange handlers for form fields